atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueService.java

```diff
@@ -1024,14 +1074,21 @@ private CfDef getCfForTable(TableReference tableRef, byte[] rawMetadata) {
                                     if (shouldTraceQuery(tableRef)) {
                                         ByteBuffer recv_trace = client.trace_next_query();
                                         Stopwatch stopwatch = Stopwatch.createStarted();
-                                        firstPage = client.get_range_slices(colFam, pred, keyRange, consistency);
-                                        long duration = stopwatch.elapsed(TimeUnit.MILLISECONDS);
-                                        if (duration > getMinimumDurationToTraceMillis()) {
-                                            log.error("Traced a call to " + tableRef.getQualifiedName() + " that took " + duration + " ms."
-                                                    + " It will appear in system_traces with UUID=" + CassandraKeyValueServices.convertCassandraByteBufferUUIDtoString(recv_trace));
+                                        try {
+                                            firstPage = client.get_range_slices(colFam, pred, keyRange, consistency);
+                                        } catch (Exception e) {
+                                            logFailedCall(tableRef);
+                                            logTraceResults(stopwatch.elapsed(TimeUnit.MILLISECONDS), tableRef, recv_trace, true);
+                                            throw e;
                                         }
+                                        logTraceResults(stopwatch.elapsed(TimeUnit.MILLISECONDS), tableRef, recv_trace, false);
                                     } else {
-                                        firstPage = client.get_range_slices(colFam, pred, keyRange, consistency);
```

Comment:
generally I hate how one of these looks should probably create one method that does this, pass the actual client call as a function to that and do it that way